ch.sbb.scion.rcp.workbench/src/ch/sbb/scion/rcp/workbench/view/ViewIntentInterceptor.java

```diff
@@ -62,43 +62,55 @@ public boolean handle(IntentMessage<Map<String, ?>> intentMessage) throws PartIn
     return false;
   }
 
-  private void openMicrofrontendEditor(IntentMessage<Map<String, ?>> intentMessage) throws PartInitException {
+  private void openMicrofrontendEditor(final IntentMessage<Map<String, ?>> intentMessage) throws PartInitException {
     var activePage = getActivePage();
     var capability = intentMessage.capability;
     var intent = intentMessage.intent;
-    var target = Optional.ofNullable(intentMessage.body).map(body -> (String) body.get("target")).orElse("default");
+    var target = Optional.ofNullable(intentMessage.body).map(body -> (String) body.get("target")).orElse("auto");
+    // TODO [ISW] Eclipse RCP ignores the `activate` flag passed to IWorkbenchPage.openEditor. The editor is always activated even if passing `false`.
+    var activate = Optional.ofNullable(intentMessage.body).map(body -> (Boolean) body.get("activate")).orElse(Boolean.TRUE).booleanValue();
 
     switch (target) {
-      case "blank": {
-        activePage.openEditor(new MicrofrontendViewEditorInput(capability, intent), MicrofrontendViewEditorPart.ID, true,
+    case "blank": {
+      // Open new editor.
+      activePage.openEditor(new MicrofrontendViewEditorInput(capability, intent), MicrofrontendViewEditorPart.ID, activate,
+          IWorkbenchPage.MATCH_NONE);
+      return;
+    }
+    case "auto": {
+      // Activate editor if already opened, otherwise open a new editor.
+      var editorRef = findMicrofrontendViewEditor(capability, intent);
+      var editor = editorRef != null ? (IReusableEditor) editorRef.getEditor(true) : null;
+      if (editor == null || editorRef == null) {

```

Comment:
I think the editorRef null check can be removed. If the editor is not null, then the editorRef cannot be null, otherwise the editor would be null.

Reply:
You are absolutely right, but the Eclipse compiler complains about it. If you like, I can remove the redundant check, but then we will get a lint warning.